JavaFXUserInterface/src/main/java/external/ModelDAOImpl.java

```diff
@@ -106,6 +116,13 @@ public void updateRMSE(Double rmse, int modelId, int assetTypeId) {
                     newModel.setModelID(rs.getInt("model_id"));
                     newModel.setModelName(rs.getString("name"));
                     newModel.setDescription(rs.getString("description"));
+                    try (PreparedStatement pss = getConnection().prepareStatement(GET_LATEST_RMSE)) {
+                        pss.setInt(1, newModel.getModelID());
+                        pss.setString(2, String.valueOf(assetTypeID));
+                        try (ResultSet rss = pss.executeQuery()) {
+                            if (rss.next())
+                                newModel.setRMSE(rss.getString("rmse"));
+                        }}
```

Comment:
Avoid doing 1 query per model, simply get the trainedmodels with status id 2 for the specified asset type id and then iterate over that list.. you will then only need 1 query instead of 9...

Reply:
changes applied